skyportal/handlers/api/observation_plan.py

```diff
@@ -641,3 +648,169 @@ async def get(self, observation_plan_request_id):
+    @auth_or_token
+    def post(self, observation_plan_request_id):
+        """
+        ---
+        description: Submit the observation plan to treasuremap.space
+        tags:
+          - observation_plan_requests
+        parameters:
+          - in: path
+            name: observation_plan_id
+            required: true
+            schema:
+              type: string
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        options = [
+            joinedload(ObservationPlanRequest.observation_plans)
+            .joinedload(EventObservationPlan.planned_observations)
+            .joinedload(PlannedObservation.field)
+        ]
+
+        observation_plan_request = ObservationPlanRequest.get_if_accessible_by(
+            observation_plan_request_id,
+            self.current_user,
+            mode="read",
+            raise_if_none=True,
+            options=options,
+        )
+        self.verify_and_commit()
+
+        event = (
+            GcnEvent.query_records_accessible_by(
+                self.current_user,
+                options=[
+                    joinedload(GcnEvent.gcn_notices),
+                ],
+            )
+            .filter(GcnEvent.id == observation_plan_request.gcnevent_id)
+            .first()
+        )
+
+        allocation = Allocation.get_if_accessible_by(
+            observation_plan_request.allocation_id,
+            self.current_user,
+            raise_if_none=True,
+        )
+
+        instrument = allocation.instrument
+
+        altdata = allocation.altdata
+        if not altdata:
+            raise self.error('Missing allocation information.')
+
+        observation_plan = observation_plan_request.observation_plans[0]
+        planned_observations = observation_plan.planned_observations
+
+        if len(planned_observations) == 0:
+            return self.error('Cannot submit observing plan with no observations.')
+
+        graceid = event.graceid
+        payload = {"graceid": graceid, "api_token": altdata['TREASUREMAP_API_TOKEN']}
+
+        pointings = []
+        for obs in planned_observations:
+            pointing = {}
+            pointing["ra"] = obs.field.ra
+            pointing["dec"] = obs.field.dec
+            pointing["band"] = obs.filt
+            pointing["instrumentid"] = str(instrument.treasuremap_id)
+            pointing["status"] = "planned"
+            pointing["time"] = Time(obs.obstime, format='datetime').isot
+            pointing["depth"] = 0.0
+            pointing["depth_unit"] = "ab_mag"
+            pointings.append(pointing)
+
+        payload["pointings"] = pointings
+
+        url = urllib.parse.urljoin(TREASUREMAP_URL, 'api/v0/pointings')
+        r = requests.post(url=url, json=payload)
+        r.raise_for_status()
```

Comment:
Should this be caught and converted to `self.error`?

Reply:
Here too.